### ISSUE TITLE
[3.2.3 BC break] EmptyStatement.DetectedCATCH renamed to EmptyStatement.DetectedCatch, breaking excludes

### ISSUE THREAD

**Issue by Majkl578** [squizlabs/PHP_CodeSniffer#1909]
The changes in #1890 introduced a BC break in 3.2.3 bugfix release.

The following code did not produce any error in 3.2.2:
```php
<?php


try {
    doSomething();
} catch (SomeException $e) {
    // ignore
}
```
It is now producing the following error in 3.2.3:
```
FILE: test.php
------------------------------------------------------------------------------------------------
FOUND 1 ERROR AFFECTING 1 LINE
------------------------------------------------------------------------------------------------
 6 | ERROR | Empty CATCH statement detected (Generic.CodeAnalysis.EmptyStatement.DetectedCatch)
------------------------------------------------------------------------------------------------
```

**Comment by photodude** [comment]
I think you will find the issue is related to the Squiz.Commenting.EmptyCatchComment sniff

**Comment by Majkl578** [comment]
As you can see from the error message, the sniff error code is `Generic.CodeAnalysis.EmptyStatement.DetectedCatch`.

**Comment by Majkl578** [comment]
Ok, I found the culprit: it is caused by the change to the error code.
3.2.2: `Generic.CodeAnalysis.EmptyStatement.DetectedCatch`
3.2.3: `Generic.CodeAnalysis.EmptyStatement.DetectedCATCH`

We have an exclude for this specific error which is no longer working.
Rewording the issue, but keeping the BC break label.

**Comment by KacerCZ** [comment]
This BC break was introduced by commit https://github.com/squizlabs/PHP_CodeSniffer/commit/eefb31c68e4c9e4c390d08859163ca7e587ca46d

**Comment by photodude** [comment]
If you look at the unit test change you can see a finally block added after the try/catch. I suspect if you add just a try/catch to the unit test it should confirm the issue. (I.e. There probably should be a try/catch and a try/catch/finally in the unit test)

Should be easier to pin down what part of the change is the issue that way.

**Comment by jrfnl** [comment]
> Should be easier to pin down what part of the change is the issue that way.

The change which is the issue is in line 90/91 of `EmptyStatementSniff` which now does a `ucfirst(strtolower($name))` for the errorcode.

**Comment by photodude** [comment]
@gsherwood unfortunately that change also made it over to 2.9.x with the backport in #1892

**Comment by gauthierm** [comment]
Should I update my ruleset to use the new `EmptyStatementSniff.DetectedCatch` or should I wait for a new patch release of phpcs?

**Comment by gsherwood** [comment]
@gauthierm That change is now in the changelog, so I wont revert it. Please update your ruleset. Again, I apologise for that change.

**Comment by photodude** [comment]
@gsherwood I see the fix in #1914 for the Squiz ruleset `Squiz.PHP.CommentedOutCode` but no fix for the `Generic.CodeAnalysis.EmptyStatement.DetectedCatch` error noted here.

**Comment by gsherwood** [comment]
> but no fix for the Generic.CodeAnalysis.EmptyStatement.DetectedCatch error noted here

The error isn't new. The error code changed. I have added that note to the changelog so it is now official and people can use the updated codes. I have not had time to try and make the rules case insensitive.

**Comment by gsherwood** [comment]
I took a look at making sniff codes case insensitive. It's very possible, but I'm really worried about missing something and breaking rulesets because I had to change a lot of places. I've created a new issue (#1955) for this but wont implement it until version 4.
